wpiutil/src/main/native/include/wpi/mDNSAnnouncer.h

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#pragma once
+
+#include <memory>
+#include <string>
+#include <string_view>
+#include <utility>
+
+#include "wpi/span.h"
+
+namespace wpi {
+class mDNSAnnouncer {
+ public:
+  mDNSAnnouncer(std::string_view serviceName, std::string_view serviceType,
+                int port,
+                wpi::span<const std::pair<std::string, std::string>> txt);
+  ~mDNSAnnouncer() noexcept;
+
```

Comment:
The top level API should provide a way to know if there is in fact an implementation (eg to report a warning or disable user-facing functionality).  This is particularly useful in the discovery case, but should be in both.

Reply:
Done